src/api/pagination.ts

```diff
@@ -25,16 +25,176 @@ export function parsePagingQueryInput(val: any) {
   return parsedInput;
 }
 
-interface ParseLimitQueryParams {
-  maxItems: number;
-  errorMsg: string;
-}
+const MAX_BLOCKS_PER_REQUEST = 30;
+const MAX_TX_PER_REQUEST = 50;
+const MAX_ASSETS_PER_REQUEST = 50;
+const MAX_STX_INBOUND_PER_REQUEST = 500;
+const CONTRACT_MAX_EVENTS_PER_REQUEST = 50;
+const MAX_MICROBLOCKS_PER_REQUEST = 200;
+const MAX_TXS_PER_REQUEST = 200;
+const MAX_MEMPOOL_TXS_PER_REQUEST = 200;
+const TX_MAX_EVENTS_PER_REQUEST = 200;
+const MAX_TOKENS_PER_REQUEST = 200;
+
+const pagedApiRoutes = [
+  '/block',
+  '/address/:principal/transactions',
+  '/address/:stx_address/assets',
+  '/address/:stx_address/nft_events',
+  '/address/:stx_address/transactions_with_transfers',
+  '/address/:address/mempool',
+  '/address/:stx_address/stx_inbound',
+  '/burnchain/reward_slot_holders',
+  '/burnchain/reward_slot_holders/:address',
+  '/burnchain/rewards',
+  '/burnchain/rewards/:address',
+  '/contract/by_trait',
+  '/contract/:contract_id/events',
+  '/microblock',
+  '/tx',
+  '/tx/mempool/dropped',
+  '/tx/mempool',
+  '/tx/multiple',
+  '/tx/events',
+  '/tx/:tx_id',
+  '/tx/block/:block_hash',
+  '/tx/block_height/:height',
+  '/mempool',
+  '/tokens/nft/holdings',
+  '/tokens/nft/history',
+  '/tokens/nft/mints',
+  '/tokens/ft/metadata',
+  '/tokens/nft/metadata',
+] as const;
+
+type PagedApiRoutes = typeof pagedApiRoutes[number];
 
-export function parseLimitQuery({ maxItems, errorMsg }: ParseLimitQueryParams) {
-  return (val: any) => {
-    const limit = parsePagingQueryInput(val);
-    if (limit > maxItems)
-      throw new InvalidRequestError(errorMsg, InvalidRequestErrorType.invalid_query);
-    return limit;
-  };
+const pagingQueryLimits: Record<PagedApiRoutes, { defaultLimit: number; maxLimit: number }> = {
```

Comment:
I think instead of keeping limits per endpoint, they should be per "return type" e.g.:
* If the endpoint is returning blocks, default limit should be 10
* If it returns transactions, default should be 20
* etc.

Reply:
Hey I have a branch here that implements your feedback (though I haven't changed the calls to get the limits yet. That will be simple and the calls will look pretty much identical). Could you take a look? A few endpoints collapsed nicely into resource categories, but a lot didn't. If one where to add a new endpoint it's not obvious for the resource type approach what default and max limit should be used. Additionally, using an endpoint approach, there's more flexibility to use whatever default and limit we want. I think we should stick with the endpoint approach.